### ISSUE TITLE
Allow drag selection inside nodes

### ISSUE THREAD

**Issue by hiepnguyen3001** [xyflow/xyflow#4280]
### Please describe the feature that you want to propose

Currently, it is not possible to start a drag selection box inside a node. May I suggest a new property like `enableDragSelect`? 

I'm aware this behavior means that you cannot drag the node itself anymore. But in that case, we can make it so that drag selection is only enabled inside node when it has a specified `dragHandle`. Then both dragging node by its drag handle and drag selection inside node is possible.

**Comment by bcakmakoglu** [comment]
[You might wanna read this section of the docs.](https://reactflow.dev/api-reference/types/node-props#notes)

TL;DR: add the `nodrag` class name to elements that should not initiate a drag action inside a node.

**Comment by hiepnguyen3001** [comment]
But that only prevents dragging the node, and you still cannot start drag selection from inside the node.

**Comment by bcakmakoglu** [comment]
Well, add some css to the element `user-select: text;` or `all` and you're good.

**Comment by hiepnguyen3001** [comment]
I did and it still doesn't work.
https://github.com/xyflow/xyflow/assets/139328005/80090cee-bace-43c4-9e83-627abc063eee

**Comment by bcakmakoglu** [comment]
Mh... not quite sure what the video is supposed to tell me? What exactly are you trying to achieve with "dragging" here?

**Comment by hiepnguyen3001** [comment]
In the video I start dragging outside of the box it triggers drag selection but when I do it from inside the box nothing happens.

**Comment by bcakmakoglu** [comment]
Aha! Well that's quite different from the original issue I assumed you had (thought you were trying to use some input text selection or whatever inside of a node, which the above class name + css would've allowed).
So the problem is using the RF built-in selection box while having the mouse on a node, is that correct?

**Comment by bcakmakoglu** [comment]
If that's the case, I don't think that works out of the box sadly - so your initial request might make sense although that's for the maintainers to decide and not me 😄 

At least we were able to clarify what the *exact* issue here is ^^

**Comment by moklick** [comment]
That's a good question @hiepnguyen3001. Unfortunately this is only possible with a workaround. You could listen to the selection key and set `nodesDraggable={false}` for example. Would that work?

**Comment by hiepnguyen3001** [comment]
> That's a good question @hiepnguyen3001. Unfortunately this is only possible with a workaround. You could listen to the selection key and set `nodesDraggable={false}` for example. Would that work?

I don't think so @moklick. Even when I set `nodesDraggable={false}`, it's still not possible to start a selection box from inside a node.

**Comment by moklick** [comment]
Damn.. The only way I see then is to add a class or set a style directly that prevents all pointer events.

**Comment by MatheusRoichman** [comment]
`pointer-events: none;` on the node that you want to start the drag selection in worked for me.

**Comment by heavy-d** [comment]
We needed this as well and found the same workaround with pointer-events: none;
You can still have other elements inside your Group node that do have pointer events.

Basically we wanted the inner part of a Group node to act more like the Pane.
Additionaly, we wanted the Group node itself to be not selectable by default when using the selection rectangle.
Our hacky solution for that was;

```
  useEffect(() => {
    // Selectable group nodes only when spacekey is pressed
     if (spaceKeyPressed) {
      updateNode(props.id, { selectable: true });
    } else {
      updateNode(props.id, { selectable: false });
    }
  }, [updateNode, props.id, spaceKeyPressed]);

```
// note: we use a custom store for the nodes, so the reactflow way looks a bit different

There are more things that could still be improved with Group nodes, but the recent update was already nice to see.

**Comment by moklick** [comment]
If you need this feature, the workaround is to set `pointer-events: none`.
